diff -uraN xfsprogs-6.18.0/mkfs/xfs_mkfs.c omv-xfsprogs-6.18.0/mkfs/xfs_mkfs.c
--- xfsprogs-6.18.0/mkfs/xfs_mkfs.c	2025-12-19 15:01:29.000000000 +0100
+++ omv-xfsprogs-6.18.0/mkfs/xfs_mkfs.c	2026-01-03 02:20:45.342481040 +0100
@@ -5986,7 +5986,7 @@
 			.nodalign = false,
 			.nortalign = false,
 			.bigtime = true,
-			.nrext64 = true,
+			.nrext64 = false,
 			.exchrange = true,
 			/*
 			 * When we decide to enable a new feature by default,
