diff -up xfsprogs-4.15.1/libxcmd/Makefile.0000~ xfsprogs-4.15.1/libxcmd/Makefile
--- xfsprogs-4.15.1/libxcmd/Makefile.0000~	2018-04-09 22:05:39.288047677 +0200
+++ xfsprogs-4.15.1/libxcmd/Makefile	2018-04-09 22:06:15.180669974 +0200
@@ -12,6 +12,8 @@ LT_AGE = 0
 
 CFILES = command.c input.c help.c quit.c
 
+LTLIBS = $(LIBXFS) $(LIBBLKID)
+
 ifeq ($(ENABLE_READLINE),yes)
 LCFLAGS += -DENABLE_READLINE
 LTLIBS += $(LIBREADLINE) $(LIBTERMCAP)
