diff -uraN xfsprogs-5.7.0/libxcmd/Makefile omv-xfsprogs-5.7.0/libxcmd/Makefile
--- xfsprogs-5.7.0/libxcmd/Makefile	2020-05-28 23:29:08.000000000 +0200
+++ omv-xfsprogs-5.7.0/libxcmd/Makefile	2020-07-25 13:58:51.685709950 +0200
@@ -14,6 +14,8 @@
 
 CFILES = command.c input.c help.c quit.c
 
+LTLIBS = $(LIBXFS) $(LIBBLKID)
+
 ifeq ($(ENABLE_EDITLINE),yes)
 LCFLAGS += -DENABLE_EDITLINE
 LTLIBS += $(LIBEDITLINE) $(LIBTERMCAP)
